opam-ci-check/bin/main.ml

```diff
@@ -155,29 +125,34 @@ let lint_cmd =
   in
   Cmd.v info term
 
-let list_cmd =
-  let doc = "List the revdeps for a package" in
+let no_cache =
+  let info = Arg.info [ "no-cache" ] ~doc:"Don't use the docker cache" in
+  Arg.value (Arg.flag info)
+
+let build_test_cmd =
+  let doc = "Build and test a package" in
   let term =
     Term.(
-      const show_revdeps $ pkg_term $ local_opam_repo_term
-      $ no_transitive_revdeps)
+      const build_run_spec $ no_cache $ const BuildTest $ pkg_term
+      $ local_opam_repo_term)
     |> to_exit_code
   in
   let info =
-    Cmd.info "list" ~doc ~sdocs:"COMMON OPTIONS" ~exits:Cmd.Exit.defaults
+    Cmd.info "build-test" ~doc ~sdocs:"COMMON OPTIONS" ~exits:Cmd.Exit.defaults
   in
   Cmd.v info term
 
-let test_cmd =
-  let doc = "Test the revdeps for a package" in
+let lower_bounds_cmd =
+  let doc = "Build a package with lower bounds of dependencies" in
   let term =
     Term.(
-      const test_revdeps $ pkg_term $ local_opam_repo_term $ use_dune_term
-      $ no_transitive_revdeps)
+      const build_run_spec $ no_cache $ const BuildLowerBounds $ pkg_term
+      $ local_opam_repo_term)
     |> to_exit_code
   in
   let info =
-    Cmd.info "test" ~doc ~sdocs:"COMMON OPTIONS" ~exits:Cmd.Exit.defaults
+    Cmd.info "lower-bounds" ~doc ~sdocs:"COMMON OPTIONS"
+      ~exits:Cmd.Exit.defaults
```

Comment:
Al alternative interface would give us a CLI like:

```
opam-ci-check build [--test-test] [--lower-bounds]
```

This would simplify the execution logic when constructing the build config, because the booleans from the flags could just be passed to `Spec.opam` directly, and the `build_spec` type would not be needed. It would also allow taking advantage of the ability to generate specs that have both flags set to true.

WDYT?

Reply:
This is a great suggestion. I've implemented it, thanks!